build.gradle

```diff
@@ -148,33 +149,60 @@ tasks.register('checkAkitInstall', JavaExec) {
     classpath = sourceSets.main.runtimeClasspath
 }
 
-tasks.register('installLocalGitHookPostCheckout', Copy) {
-    from new File(rootProject.rootDir, '.scripts/post-checkout')
-    into { new File(rootProject.rootDir, '.git/hooks') }
-    filePermissions {
-        unix(0775)
-    }
-    doLast {
-        println("---Added Post Checkout Git Hook Script---")
+def ensureSymlink(link, target) {
+    if (!Files.isSymbolicLink(link)) {
+        if (Files.exists(link)) {
+            Files.delete(link)
+        }
+
+        Files.createSymbolicLink(link, target)
+
+        def relative = { f -> rootDir.toPath().relativize(f) }
+        println("--- Installed Symbolic Link ${relative(link)} -> ${relative(target)} ---")
     }
 }
 
-tasks.register('installLocalGitHookPreCommit', Copy) {
-    from new File(rootProject.rootDir, '.scripts/pre-commit')
-    into { new File(rootProject.rootDir, '.git/hooks') }
-    filePermissions {
-        unix(0775)
-    }
+// Setup local git settings
+def gitConfig = 'Git Configuration'
+tasks.register('installLocalGitHookSymlinks') {
+    group = gitConfig
+
+    outputs.dir "$rootDir/.git/hooks/"
+
     doLast {
-        println("---Added Pre Commit Git Hook Script---")
+        def linkDir = file("$rootDir/.git/hooks/").toPath()
+        def gitHooks = fileTree("$rootDir/.scripts/git-hooks/")
+        gitHooks.forEach { File target ->
+            def link = linkDir.resolve(target.toPath().getFileName())
+
+            ensureSymlink(link, target.toPath())
+        }
     }
 }
 
+tasks.register('configLocalGitBlameIgnoreRevsFile', Exec) {
+    group = gitConfig
+
+    workingDir rootProject.rootDir
+    commandLine 'git', 'config', '--local', 'blame.ignoreRevsFile', '.git-blame-ignore-revs'
+
+    outputs.file('.git/config')
+}
+
+def gitConfigTasks = tasks.matching { it.group == gitConfig }
+tasks
+        .matching { it.group != gitConfig }
+        .configureEach { it.dependsOn gitConfigTasks }
```

Comment:
i don't like this space, i would do it 1 line or 2

Reply:
it just doesn't really look good with the dots on different columns, but if it really bothers you two lines is certainly more readable than one

Reply:
so 2 please